src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesArgs.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Commands
+{
+    public class SourcesArgs
+    {
+        public delegate void Log(string message);
+
+        public ISettings Settings { get; }
+        public IPackageSourceProvider SourceProvider { get; }
+        public SourcesAction Action { get; }
+        public string Name { get; }
+        public string Source { get; }
+        public string Username { get; }
+        public string Password { get; }
+        public bool StorePasswordInClearText { get; }
+        public string ValidAuthenticationTypes { get; }
+        public SourcesListFormat Format { get; }
+        public bool Interactive { get; }
+        public string ConfigFile { get; }
+        private bool IsQuiet { get; }
+        public Log LogError { get; }
+        public Log LogMinimal { get; }
+        public Log LogQuiet { get; }
```

Comment:
Not sure I'm following the Log pattern here. In Push, we defer all the logging to the Protocol library (PackageUpdateResource). What benefit is there to having the IsQuiet and other Log objects here, and why different than other Commands?

Reply:
thanks. testing on your machine helped me realize that "nuget sources list -verbosity quiet" shows nothing...
that was different that I had believed to be true.  As such, i'll be removing LogQuiet...LogError is going away due to the return to the exception.  But we'll still have LogMinimal for now (which respects isQuiet) --- and we'll add LogWarning in future work as well.

fixed in next commit.

Reply:
yes, i got carried away with my Logging infrastructure when I was doing this and lost the exceptions. Nice catch. Moving exception type to nuget.commands.dll in nuget.commands namespace and using it again here.
fix in next commit.